onnxruntime/core/session/inference_session.h

```diff
@@ -128,7 +135,15 @@ class InferenceSession {
     * Calling this API is optional.
     * @return OK if success.
     */
-  common::Status RegisterGraphTransformer(std::unique_ptr<onnxruntime::GraphTransformer> p_graph_transformer);
+  common::Status RegisterGraphTransformer(std::unique_ptr<onnxruntime::GraphTransformer> p_graph_transformer, std::vector<std::string>&& providers);
```

Comment:
>std::vector<std::string>&& providers [](start = 110, length = 36)

Do we need to force a caller to provider an rvalue reference for 'providers'? Seems like just taking a reference and copying the names wouldn't incur much cost and be a friendlier API.